### ISSUE TITLE
Connection error

### ISSUE THREAD

**Issue by katjabruhn** [avocode/avocode#960]
Hello, after synchronizing my Sketch file with Avocode I would open the artboards in Avocode, but I always get a connection error / download error. What can I do? I have the latest Avocode version and the latest sketch plugin (3.7).

Please, can you help me.

**Comment by connor-bean** [comment]
Hey, Katja -

Can you send us a screenshot of the error you're seeing? What OS are you running? And are you running Sketch Beta? 

Thanks,
Connor, Avocode Support

**Comment by katjabruhn** [comment]
Hey Connor,

I am sending you the requested error messages. I’m running OS X El Capitan Version 10.11.2 and Sketch Version 3.8.3.

Best regards,
Katja

Von: Connor <notifications@github.com>
Antworten an: avocode/avocode <reply@reply.github.com>
Datum: Montag, 4. Juli 2016 um 15:30
An: avocode/avocode <avocode@noreply.github.com>
Cc: Katja Bruhn <Katja.Bruhn@cosmoblonde.de>, Author <author@noreply.github.com>
Betreff: Re: [avocode/avocode] Connection error (#960)


Hey, Katja -

Can you send us a screenshot of the error you're seeing? What OS are you running? And are you running Sketch Beta?

Thanks,
Connor, Avocode Support

**Comment by connor-bean** [comment]
Hey, Katja -

Can you try resending the screenshots? Or send them to team@avocode.com? I don't see them in Github.

Thanks!
Connor, Avocode Support

**Comment by katjabruhn** [comment]
Hey Connor,
here are the screenshots: 

![bildschirmfoto 2016-07-04 um 15 33 30](https://cloud.githubusercontent.com/assets/18698929/16582595/14e695f8-42b0-11e6-8ae8-8ebee6e01de0.png)
![bildschirmfoto 2016-07-04 um 15 33 33](https://cloud.githubusercontent.com/assets/18698929/16582596/1507cee4-42b0-11e6-9641-42ff0aafc7a6.png)

**Comment by vojtatranta** [comment]
Hi @katjabruhn,
can you for example drag and drop a .psd into Avocode to find out whether it is just a problem of sketch?

**Comment by katjabruhn** [comment]
Hi @vojtatranta , it seems to be just a problem with sketch-files. I tested it with a psd file and it works. But is the attached message may be related to this problem?
![bildschirmfoto 2016-07-05 um 13 48 19](https://cloud.githubusercontent.com/assets/18698929/16583707/9c65367c-42b7-11e6-80b1-c95aeeb9ef6f.png)

**Comment by vojtatranta** [comment]
@katjabruhn aha, I see, what version of Sketch and Avocode Sketch Extension do you use?

**Comment by katjabruhn** [comment]
sketch version 3.8.3
avocode sketch plugin versions 3.7

**Comment by connor-bean** [comment]
Hey, @katjabruhn -

Can you send us the problematic design to team@avocode.com?

Thanks,
Connor, Avocode Support

**Comment by katjabruhn** [comment]
Hello, I uninstalled sketch and installed it again. I synced the Sketch file with Avocode and then the following error occurred. Can you now see what's causing the error?


IMPORTANT: Please try to describe how did it happen.

-----
App v.2.8.4
Source UID: 17351
```
Error: Could not call remote function 'quit'. Check that the function signature is correct. Underlying error: Attempting to call a function in a renderer window that has been closed or released. Function provided here: avocode.min.js:23:13547.
Error: Could not call remote function 'quit'. Check that the function signature is correct. Underlying error: Attempting to call a function in a renderer window that has been closed or released. Function provided here: avocode.min.js:23:13547.
    at callFunction (/Applications/Avocode.app/Contents/Resources/electron.asar/browser/rpc-server.js:222:11)
    at EventEmitter.<anonymous> (/Applications/Avocode.app/Contents/Resources/electron.asar/browser/rpc-server.js:303:5)
    at emitMany (events.js:127:13)
    at EventEmitter.emit (events.js:201:7)
    at EventEmitter.<anonymous> (/Applications/Avocode.app/Contents/Resources/electron.asar/browser/api/web-contents.js:165:13)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at metaToValue (/Applications/Avocode.app/Contents/Resources/electron.asar/renderer/api/remote.js:178:13)
    at remoteMemberFunction (/Applications/Avocode.app/Contents/Resources/electron.asar/renderer/api/remote.js:112:18)
    at t.quit (file:///Applications/Avocode.app/Contents/Resources/app.asar/dist/avocode.min.js:2:7270)
    at a.call.concat.click (file:///Applications/Avocode.app/Contents/Resources/app.asar/dist/avocode.min.js:25:14416)
    at CallbacksRegistry.apply (/Applications/Avocode.app/Contents/Resources/electron.asar/common/api/callbacks-registry.js:54:39)
    at EventEmitter.<anonymous> (/Applications/Avocode.app/Contents/Resources/electron.asar/renderer/api/remote.js:243:21)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

**Comment by greystate** [comment]
I'm seeing the "Connection error" problems too with a specific Sketch file today (I've sent it to team@avocode.com)

**Comment by greystate** [comment]
Just to keep you updated - I still have this problem for that specific design, in Avocode 2.9.1

**Comment by Zalrak** [comment]
Got the same connection error running Avocode on Ubuntu 16.04, however found out that it was a permission problem by reading the error message below. Avocode wasn't allowed to create a folder in my system.

Solved the problem by running the app as administrator, hope this helps somebody else.

-----
App v. `2.10.1`
Source UID: 82781
```
Error: EACCES: permission denied, mkdir '/home/user/.avocode/userdata/82781/documents/7de5c04a52dd4cdfad2cd0ec3260f4e0'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:922:18)
    at Function.sync (/opt/avocode/resources/app.asar/node_modules/fs-plus/node_modules/mkdirp/index.js:55:12)
    at Object.makeTreeSync (/opt/avocode/resources/app.asar/node_modules/fs-plus/lib/fs-plus.js:372:23)
    at e.exports.e.ensureDirectory (file:///opt/avocode/resources/app.asar/dist/avocode.min.js:16:20787)
    at e._downloadRevisionMetadata (file:///opt/avocode/resources/app.asar/dist/avocode.min.js:14:5579)
    at e._downloadDesignRevision (file:///opt/avocode/resources/app.asar/dist/avocode.min.js:14:5389)
    at e._downloadRevision (file:///opt/avocode/resources/app.asar/dist/avocode.min.js:14:3877)
    at file:///opt/avocode/resources/app.asar/dist/avocode.min.js:14:3352
    at e._processRevision (file:///opt/avocode/resources/app.asar/dist/avocode.min.js:24:29152)
    at e._processRevisi
```

**Comment by connor-bean** [comment]
Hey this is now fixed in 2.14.3. Please update your app
